### PR TITLE
EOS-20069 [F-44] API to return audit logging schema

### PR DESCRIPTION
**Pull request by abhilekh268**

### CODE REVIEW

File: server/s3_get_audit_log_schema.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+void S3GetAuditLogSchema::setup_steps() {
+  set_string();
+  send_response_to_s3_client();
+}
+
+void S3GetAuditLogSchema::set_string() { schema = "efgh"; }
+
+void S3GetAuditLogSchema::send_response_to_s3_client() {
+
```
Comment (knrajnambiar76) [inline server/s3_get_audit_log_schema.cc:42]: logging -- Entry...

File: server/s3_get_audit_log_schema.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#include <string>
+#include "s3_request_object.h"
+#include "s3_action_base.h"
+class S3GetAuditLogSchema : public S3Action {
```
Comment (knrajnambiar76) [inline server/s3_get_audit_log_schema.h:25]: Follow the convention we have been following
ie have class named as some action class  even the file name as a *action* file

File: server/s3_management_api_handler.cc [hunk 3]
```diff
@@ -30,10 +31,17 @@ void S3ManagementAPIHandler::create_action() {
     case S3OperationCode::none:
       // Perform operation on Service.
       switch (request->http_verb()) {
-        case S3HttpVerb::DELETE:
+        case S3HttpVerb::DELETE: {
           action = std::make_shared<S3AccountDeleteMetadataAction>(request);
           s3_log(S3_LOG_DEBUG, request_id, "S3AccountDeleteMetadataAction");
-          break;
+        } break;
```
Comment (knrajnambiar76) [inline server/s3_management_api_handler.cc:37]: break above } ?

File: server/s3_get_audit_log_schema.cc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+#include "s3_error_codes.h"
+#include "s3_log.h"
+#include "s3_api_handler.h"
+#include "s3_get_audit_log_schema.h"
+
+S3GetAuditLogSchema::S3GetAuditLogSchema(std::shared_ptr<S3RequestObject> req)
+    : S3Action(req, true, nullptr, true, true), request(req) {
+  schema = "abc";
```
Comment (knrajnambiar76) [inline server/s3_get_audit_log_schema.cc:28]: You can initialize with ""

**Review by bkirunge7 — COMMENTED** [review]
looks good, pls go ahead

File: server/s3_get_audit_log_schema.cc [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+}
+S3GetAuditLogSchema::~S3GetAuditLogSchema() {
+  s3_log(S3_LOG_DEBUG, "", "%s\n", __func__);
+}
+void S3GetAuditLogSchema::setup_steps() {
+  set_string();
+  send_response_to_s3_client();
+}
+
+void S3GetAuditLogSchema::set_string() { schema = "efgh"; }
```
Comment (t7ko-seagate) [inline server/s3_get_audit_log_schema.cc:39]: Is this some placeholder?  or this is the actual final string?  If placeholder, better put it so it's immediately noticeable, e.g. "PLACEHOLDER-efgh".  And add a comment.

File: server/s3_management_api_handler.cc (hunk 3 above)
Comment (t7ko-seagate) [reply server/s3_management_api_handler.cc:37]: why do we need {} here? it's not rquired and I dont think we use braces in rest of code inside `switch` statements...

**Review by t7ko-seagate — CHANGES_REQUESTED** [review]

File: server/s3_get_audit_log_schema.cc [hunk 6]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (amit-macrometa) [inline server/s3_get_audit_log_schema.cc:2]: 2021

File: server/s3_management_api_handler.cc (hunk 3 above)
Comment (abhilekh268) [reply server/s3_management_api_handler.cc:37]: to resolve issue of not handling post put enumaration values,i had to add this {},which resolved the issue.

File: server/s3_get_audit_log_schema.cc (hunk 1 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema.cc]: added.

File: server/s3_get_audit_log_schema.h (hunk 2 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema.h]: added

File: server/s3_get_audit_log_schema.cc (hunk 4 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema.cc]: done

File: server/s3_get_audit_log_schema.cc (hunk 5 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema.cc]: updated with final thing

File: server/s3_get_audit_log_schema.cc (hunk 6 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema.cc]: done

File: server/s3_get_audit_log_schema_action.cc [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+}
+S3GetAuditLogSchemaAction::~S3GetAuditLogSchemaAction() {
+  s3_log(S3_LOG_DEBUG, "", "%s\n", __func__);
+}
+void S3GetAuditLogSchemaAction::setup_steps() {
+  set_string();
+  send_response_to_s3_client();
+}
+
+void S3GetAuditLogSchemaAction::set_string() {
```
Comment (bkirunge7) [inline server/s3_get_audit_log_schema_action.cc:40]: can we read this json from file

File: server/s3_get_audit_log_schema_action.cc [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+      "\"filterable\":false},{\"field_id\":\"user_agent\",\"label\":\"User "
+      "Agent\",\"display_id\":123,\"display\":true,\"sortable\":true,"
+      "\"filterable\":false},{\"field_id\":\"version_id\",\"label\":\"Version "
+      "Id\",\"display_id\":124,\"display\":true,\"sortable\":true,"
+      "\"filterable\":false}]";
+}
+
+void S3GetAuditLogSchemaAction::send_response_to_s3_client() {
+  s3_log(S3_LOG_DEBUG, "", "%s Entry \n", __func__);
+  request->set_out_header_value("csmschema", schema.c_str());
```
Comment (bkirunge7) [inline server/s3_get_audit_log_schema_action.cc:99]: rename csmschema, auditlog_ui_layout

File: server/s3_management_api_handler.cc [hunk 9]
```diff
@@ -30,10 +31,17 @@ void S3ManagementAPIHandler::create_action() {
     case S3OperationCode::none:
```
Comment (bkirunge7) [inline server/s3_management_api_handler.cc:31]: Add a system test

**Review by knrajnambiar76 — APPROVED** [review]

**Review by t7ko-seagate — APPROVED** [review]
Adding +1 but please address the comment from Basav.

File: server/s3_get_audit_log_schema_action.cc (hunk 7 above)
Comment (abhilekh268) [reply server/s3_get_audit_log_schema_action.cc:44]: done

File: server/s3_management_api_handler.cc (hunk 9 above)
Comment (abhilekh268) [reply server/s3_management_api_handler.cc:31]: added

**Review by bkirunge7 — APPROVED** [review]

**Comment by nileshgovande** [comment]
Mini provisioner pass job - http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1426/

**Pull request merged**